docs/pages/model_hub/asr.md

```diff
@@ -49,4 +49,68 @@ Here are some other possible configurations for the Whisper deployment:
+
+Example configuration for the PyannoteSpeakerDiarization model is available at [Speaker Diarization](./speaker_recognition.md) model hub.
+
+You can simply define the model deployments and the endpoint to transcribe the video with diarization. Below code snippet shows the custom endpoint class `TranscribeVideoWithDiarEndpoint` to combine the outputs from ASR and diarization deployments:
+
+    ```python
+    from aana.api.api_generation import Endpoint
+    from aana.core.models.speaker import PyannoteSpeakerDiarizationParams
+    from aana.core.models.video import VideoInput
+    from aana.core.models.whisper import WhisperParams
+    from aana.deployments.whisper_deployment import WhisperOutput
+
+    from aana.deployments.aana_deployment_handle import AanaDeploymentHandle
+
+    from aana.integrations.external.yt_dlp import download_video
+    from aana.processors.remote import run_remote
+    from aana.processors.speaker import ASRPostProcessingForDiarization
+    from aana.processors.video import extract_audio
+
+    class TranscribeVideoWithDiarEndpoint(Endpoint):
+    """Transcribe video with diarization endpoint."""
+
+    async def initialize(self):
+        """Initialize the endpoint."""
+        self.asr_handle = await AanaDeploymentHandle.create("asr_deployment")
+        self.diar_handle = await AanaDeploymentHandle.create("diarization_deployment")
+        await super().initialize()
+
+    async def run(
+        self,
+        video: VideoInput,
+        whisper_params: WhisperParams,
+        diar_params: PyannoteSpeakerDiarizationParams,
+    ) -> WhisperOutput:
+        """Transcribe video with diarization."""
+        video_obj = await run_remote(download_video)(video_input=video)
+        audio = extract_audio(video=video_obj)
+
+        # diarized transcript requires word_timestamps from ASR
+        whisper_params.word_timestamps = True
+        transcription = await self.asr_handle.transcribe(
+            audio=audio, params=whisper_params
+        )
+        diarized_output = await self.diar_handle.diarize(
+            audio=audio, params=diar_params
+        )
+        post_processor = ASRPostProcessingForDiarization(
+            diarized_segments=diarized_output["segments"],
+            transcription_segments=transcription["segments"],
+        )
+        updated_segments = post_processor.process()
+        output_segments = [
+            s.model_dump(include=["text", "time_interval", "speaker"])
+            for s in updated_segments
+        ]
+
+        return {"segments": output_segments}
+    ```
+An example notebook on diarized transcription is available at `notebooks/diarized_transcription_example.ipynb`.
```

Comment:
Add a link to the notebook.

Reply:
Done, it will show once merged to main as the link is: `https://github.com/mobiusml/aana_sdk/tree/main/notebooks/diarized_transcription_example.ipynb`